src/lib/metrics.rs

```diff
@@ -174,8 +174,15 @@ impl UnmatchedCounter {
     }
 
     /// Write the top `n` unmatched barcodes to a `most_frequent_unmatched.tsv` file in the specified directory.
-    pub fn to_file<P: AsRef<Path>>(self, output_dir: P, n: usize) -> Result<()> {
-        let output_path = output_dir.as_ref().join("most_frequent_unmatched.tsv");
+    pub fn to_file<P: AsRef<Path>>(
+        self,
+        output_dir: P,
+        n: usize,
+        prefix: Option<String>,
```

Comment:
there's got to be a better way to pass this around than `Option<String>` so I can concatenate below.